### ISSUE TITLE
False positive in pattern attribute assignment: [syntax-errors] Multiple assignments in case pattern

### ISSUE THREAD

**Issue by sasanjac** [astral-sh/ruff#17207]
### Summary

The current release 0.11.3 in preview mode throws a false positive syntax error in the following case:

`ruff check file`

```python
match foo:
    case Bar(baz=baz): # [syntax-errors] Multiple assignments in case pattern
        foobar(baz)
```

As far as I know, this is valid syntax (see PEP 622) and necessary to match against class attributes of classes that don't explicitly specify match attributes.

### Version

ruff 0.11.3

**Comment by ntBre** [comment]
Thanks for the report! This is the same as #17181.

**Comment by sasanjac** [comment]
oops, sorry
